tests/test_analyze.py

```diff
@@ -0,0 +1,274 @@
+
+    def test_analyze_eval_multiple_detectors(self):
+        test_data = self.create_test_data()
+        headers = ["context_docs", "user_query", "prompt"]
+        config = {
+            'hallucination': {'detector_name': 'default'},
+            'toxicity': {'detector_name': 'default'}
+        }
+        application = Application(name=self.app_name)
+        model = Model(name=self.model_name, model_type="text")
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            dec = AnalyzeEval(
+                application=application,
+                model=model,
+                evaluation_name=self.evaluation_name+"_multi",
+                dataset_collection_name=self.collection_name,
+                headers=headers,
+                api_key=self.api_key,
+                config=config
+            )
+            @dec
+            def dummy_func(context_docs, user_query, prompt):
+                return f"output: {context_docs[0]}, {user_query}, {prompt}"
+            results = dummy_func()
+            assert isinstance(results, list)
+            assert len(results) == 2
+            for output, response in results:
+                assert output.startswith("output:")
+                assert hasattr(response, 'status') or hasattr(response, 'hallucination')
+                # Do not check for hallucination/toxicity fields, just log response
+                print(f"RESPONSE: {response}")
+            assert any("deprecated" in str(warn.message).lower() for warn in w)
+
+    def test_analyze_eval_minimal_headers(self):
+        test_data = self.create_test_data(minimal_headers=True)
+        headers = ["context_docs", "user_query"]
+        config = {'hallucination': {'detector_name': 'default'}}
+        application = Application(name=self.app_name)
+        model = Model(name=self.model_name, model_type="text")
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            dec = AnalyzeEval(
+                application=application,
+                model=model,
+                evaluation_name=self.evaluation_name+"_minimal",
+                dataset_collection_name=self.collection_name,
+                headers=headers,
+                api_key=self.api_key,
+                config=config
+            )
+            @dec
+            def dummy_func(context_docs, user_query):
+                return f"output: {context_docs[0]}, {user_query}"
+            results = dummy_func()
+            assert isinstance(results, list)
+            assert len(results) == 1
+            output, response = results[0]
+            assert output.startswith("output:")
+            assert hasattr(response, 'status') or hasattr(response, 'hallucination')
+            assert any("deprecated" in str(warn.message).lower() for warn in w)
+
+    def test_analyze_eval_missing_header_error(self):
```

Comment:
This test is the only failing in the test suite. It's testing AnalyzeEval (the obsolete interface we still support). The SDK backend returns response `'task_definition must be a string'` https://github.com/aimonlabs/aimon-sdk-backend/blob/d5aa4c8cb31b7ff3a79676dad74547c92195bc03/aimon-sdk-backend/service.py#L175. Not sure why this happens, task_definition is absent from query altogether and not needed, so the backend must not have checked if it's a string.

Reply:
yes lets remove AnalyzeEval since its been deprecated for a while

Reply:
Done